File: clojure/prime-factors/src/prime_factors.clj
(ns prime-factors)

(defn of [n]
  (loop [factors [] i 2 j n]
    (cond
      (< j 2) factors
      (zero? (rem j i)) (recur (conj factors i) i (/ j i))
      :else (recur factors (inc i) j))))

File: go/prime-factors/prime_factors.go
package prime

func Factors(n int64) []int64 {
	f := []int64{}
	for i := int64(2); i <= n; i++ {
		for n%i == 0 {
			f = append(f, i)
			n /= i
		}
	}
	return f
}
